lnwallet/wallet.go

```diff
@@ -525,6 +525,19 @@ func (l *LightningWallet) handleFundingReserveRequest(req *InitFundingReserveMsg
 		return
 	}
 
+	var keyRing keychain.KeyRing = l.SecretKeyRing
+
+	// If this is a shim intent, then it may be attempting to use an
+	// existing set of keys for the funding workflow. In this case, we'll
+	// make a simple wrapper keychain.KeyRing that will proxy certain
+	// derivation calls to future callers.
+	if shimIntent, ok := fundingIntent.(*chanfunding.ShimIntent); ok {
```

Comment:
That would just be another layer of indirection which isn't even strictly needed in this case. It isn't clear what you're referring to, not what this would add (but another empty method interface). The `shimKeyRing` requires a `ShimIntent` in order to work as laid out: it passes the already agreed upon key to the wallet in order to further the funding flow. 

We can attempt to reason about all the possible ways that this may be used in the future, but they're all speculative. The only thing we need this for as is, is to enable external channel funding. The way things are set up, it also eventually enables callers to also specify their own _commitment key_ as well (since that can be entirely offline until we need to sweep our output on chain).